### PR TITLE
gormschema: supports trigger

### PR DESCRIPTION
**Pull request by luantranminh**

### CODE REVIEW

**Comment by luantranminh** [comment]
🧐 generated migration file for sqlite is missing the `END`
image

```sql
CREATE TRIGGER `trg_adding_heart_on_pet` BEFORE INSERT ON `pets` FOR EACH ROW BEGIN
	UPDATE pets SET name = name || ' ❤️' WHERE id = NEW.id;
```

I'll investigate it 🧐

**Comment by luantranminh** [comment]
trace down the bug here
https://github.com/ariga/atlas/issues/2808

File: internal/testdata/models/pet.go [hunk 1]
```diff
@@ -32,3 +34,100 @@ func (TopPetOwner) ViewDef(dialect string) []gormschema.ViewOption {
...
 	return []gormschema.ViewOption{gormschema.CreateStmt(stmt)}
 }
+
+type UserPetHistory struct {
+	UserID    uint `gorm:"primaryKey"`
+	PetID     uint `gorm:"primaryKey"`
+	CreatedAt time.Time
+}
+
+func (Pet) Triggers(dialect string) [][]gormschema.TriggerOption {
```
Comment (giautm) [inline internal/testdata/models/pet.go:44]:
```suggestion
func (Pet) Triggers() []gormschema.Trigger {
	return []gormschema.Trigger{
		gormschema.NewTrigger(gormschema.CreateStmts("RAW SQL")),
		gormschema.NewTrigger(gormschema.Name("TR_1")).
			AS("RAW SQL"),
	}
}
```

cc @a8m

Comment (a8m) [reply internal/testdata/models/pet.go:44]: I'd use keep only the raw definitions at this stage. A fluent API can be added on top of it and can be translated to the raw definition, but I don't think it's needed for now.

Comment (a8m) [reply internal/testdata/models/pet.go:44]: > gormschema.Trigger {

I like. That means we can change the ViewOptions also to be simply gormschema.View

Comment (luantranminh) [reply internal/testdata/models/pet.go]: using `gormschema.View` makes a lot of sense; I'll add it to my backlog.

**Review by giautm — APPROVED** [review]
Approved, but wait for @a8m opinion.

**Review by a8m — APPROVED** [review]

**Pull request merged**